lookout/style/format/analyzer.py

```diff
@@ -121,14 +129,7 @@ def train(cls, ptr: ReferencePointer, config: Mapping[str, Any], data_request_st
                 # workaround the check in joblib - everything still works without it
                 threading._MainThread = threading.Thread
             bscv = BayesSearchCV(
-                TrainableRules(
-                    prune_branches_algorithms=config["prune_branches_algorithms"],
-                    prune_attributes=config["prune_attributes"],
-                    top_down_greedy_budget=config["top_down_greedy_budget"],
-                    uncertain_attributes=config["uncertain_attributes"],
-                    prune_dataset_ratio=config["prune_dataset_ratio"],
-                    n_estimators=config["n_estimators"],
-                    random_state=config["random_state"]),
+                TrainableRules(config=config),
```

Comment:
This is one option (that we discussed yesterday and I'm ok with it). However I think it's silghtly better to require the config in the `FormatModel`'s `__setitem__`.

Reply:
ok, so we have it, And everything is fine, nothing should be changed? did I get you right?

Reply:
so 
`TrainableRules(**config["trainable_rules"], config=config),`

Reply:
No sorry I meant only `TrainableRules(**config["trainable_rules"])` and then `model[language] = rules, config`. I'll PR feature selection this morning in which I have something like that (it's very similar to this PR in some sense so it'd make sense to coordinate reviews on both).

Reply:
Ok, it is not a problem and I like it more, but if I remember correctly @vmarkovtsev want to store config inside rules: https://github.com/src-d/style-analyzer/blob/cfb9c2e9330bde757181537592849181420202f0/lookout/style/format/rules.py#L278

 @vmarkovtsev do you mind if we move it from Rules to just Modelforge model?

Reply:
So we shouldn't have any configs in the model. Our model is a dictionary of Rules per language. Each Rules must provide enough information to reproduce it bit-to-bit in form of a configuration dictionary. Model is super dumb and must continue to be dumb.

Some configuration bits will be the same across Rules instances and that it normal.

Reply:
ok, I will put it to model doc